From 42134c7aa02c9b3576f45664f7d349bfafb371ac Mon Sep 17 00:00:00 2001
From: Eduardo Abinader <eduardo.abinader@openbossa.org>
Date: Wed, 20 Aug 2014 07:06:50 -0400
Subject: [PATCH 2/9] wifi: Add an extra check for pending wifi data

Got another segmentation fault on chicken and egg issue; it is
also necessary to check the device pointer.

(cherry picked from commit 8beb6aa58e3bfc2fc42809032e4873e98bdbbe35)
Signed-off-by: Andreas Oberritter <obi@opendreambox.org>
---
 plugins/wifi.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/plugins/wifi.c b/plugins/wifi.c
index f5af702..bdb4c24 100644
--- a/plugins/wifi.c
+++ b/plugins/wifi.c
@@ -182,7 +182,7 @@ static struct wifi_data *get_pending_wifi_data(const char *ifname)
 		const char *dev_name;
 
 		wifi = list->data;
-		if (!wifi)
+		if (!wifi || !wifi->device)
 			continue;
 
 		dev_name = connman_device_get_string(wifi->device, "Interface");
-- 
1.9.1

